design-reference/architecture/VersionAndGo/readme.md

```diff
@@ -0,0 +1,113 @@
+# Version and Go Technical Design Purpose
+
+This design addresses Cobalt's workflow to deploy individual application infrastructure using a Cobalt template.
+
+## Definitions
+ - `Engineering Team (Eng Team)` - DevOps or Operations team consisting of Infrastructure Administrators or similar roles.
+ - `Application Team (App Team)` - Software Developers with an application to deploy to Azure.
+ - `Development Repo (Dev Repo)` - Cobalt's repo hosted on Github at [https://github.com/microsoft/cobal](https://github.com/microsoft/cobal)
+ - `Engineering Repo (Eng Repo)` - Individual template repository managed by the `Engineering Team`
+ - `Application Repo (App Repo)` - Repository that contains the application code to be deployed and managed by the `Application Team`
+ - `Version` - all versioning schemes should use [SemVer](https://semver.org/) or variant. 
+
+## Version and Go Setup Description
+Our design starts after the initial Cobalt setup, where an individual template is cloned into an organizations source control (ref: [Setup Doc](https://github.com/microsoft/cobalt/blob/master/docs/GETTING_STARTED_APP_DEV.md)). An application development team is ready to start deploying their code to infrastructure built in Azure. Typically they would make a request to the engineering team. The Eng team would provide the App team with a link to the latest infra configuration files for a specific template. The App team downloads this tar/zip file and extracts the folder/files.
+
+```
+infra/
+ |_ Application.tf
+ |_ Application.yaml
+```
+
+These files should be derived from the template folder in dev repo and cloned into the eng repo. Proposed folder path/location is:
+
+```
+infra/
+  |_template
+      |_template_name
+          |_tests
+          |_config
+          |   |_ application.tf
+          |   |_ application.yml
```

Comment:
If our goal is to use the same template file across any infrastructure deployment, might it make sense to move this to a directory that is not template specific?

Note: you can also reference templates from another repository so we may not even need app dev teams to clone this file into their repository because it would be a trivial template import: 
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#using-other-repositories

Reply:
I think the fact we can ref pipeline in another repo solves a lot of this.